lib/modelExtends.js

```diff
@@ -44,14 +48,20 @@ module.exports = function(Sequelize) {
 			}, sequelize.options.hierarchy || {}, options);
 
 			_.defaults(options, {
-				foreignKey: options.as + Utils.uppercaseFirst(options.primaryKey),
-				throughKey: this.name + Utils.uppercaseFirst(options.primaryKey),
-				throughForeignKey: Utils.singularize(options.ancestorsAs) + Utils.uppercaseFirst(options.primaryKey),
+				foreignKey: patches.underscoredIf(options.as + Utils.uppercaseFirst(options.primaryKey), underscored),
+				throughKey: patches.underscoredIf(this.name + Utils.uppercaseFirst(options.primaryKey), underscored),
+				throughForeignKey: patches.underscoredIf(Utils.singularize(options.ancestorsAs) + Utils.uppercaseFirst(options.primaryKey), underscored),
 				through: this.name + Utils.singularize(options.camelThrough ? Utils.uppercaseFirst(options.ancestorsAs) : options.ancestorsAs)
 			});
 
+			var genThroughTableName;
+			if (underscoredAll)
```

Comment:
Would you mind putting curly brackets `{}` around the if and else clauses please? (in line with rest of the code style)